src/cedict-service/cedict-data.js

```diff
@@ -0,0 +1,115 @@
+      // A dictionary's metadata
+      meta: {},
+      // An array of dictionary records
+      entries: []
+    }
+  }
+
+  /**
+   * Initializes a data object.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - A promise
+   */
+  init () {
+    return this.updateFromServer()
+  }
+
+  /**
+   * Returns one or several records from CEDICT dictionary for one or several Chinese words.
+   *
+   * @param {[string]} words - An array of Chinese words.
+   * @param {string} characterForm - A string constant that specifies
+   *        a character form in words (simplified or traditional).
+   * @returns {object} - Returns an object whose keys are the words requested and values are arrays of CEDICT records
+   *          that has those words.
+   */
+  getWords (words, characterForm) {
+    if (!words || !this._hasData) {
+      // No records can be found.
+      return []
+    }
+
+    // If a single word value is provided, convert it into an array.
+    if (!Array.isArray(words)) { words = [words] }
+
+    // Create an object with props for the words
+    let result = words.reduce((accumulator, key) => { accumulator[key] = []; return accumulator }, {}) // eslint-disable-line prefer-const
+
+    this.cedict.entries.forEach(entry => {
+      const hw = (characterForm === CedictData.characterForms.SIMPLIFIED) ? entry.simplifiedHeadword : entry.traditionalHeadword
+      words.forEach(word => {
+        if (hw === word) {
+          result[word].push(entry)
+        }
+      })
+    })
+    return result
+  }
+
+  /**
+   * Checks wither CEDICT dictionary has any data in it.
+   *
+   * @returns {boolean} - True if there is any dictionary records in the data object or false if otherwise.
+   * @private
+   */
+  get _hasData () {
+    return this.cedict.entries.length > 0
+  }
+
+  /**
+   * Loads fresh CEDICT data from a remote server.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - Returns a promise that will be resolved with undefined
+   *          if data was loaded successfully or that will be rejected with an error with data loading will fail.
+   */
+  updateFromServer () {
+    const requests = this._schema.data.chunks.map(chunk => this.loadJson(`${this._schema.data.URI}/${chunk}`))
+    return Promise.all(requests).then(chunks => {
+      this.cedict.meta = chunks[0].meta
+      this.cedict.entries = chunks.map(piece => piece.entries).flat()
+      this.isReady = true
+    })
+  }
+
+  /**
+   * Loads a single JSON file from a specified URL and decodes it.
+   *
+   * @param {string} url - A strings that specifies a URL of a JSON file
+   * @returns {Promise<object>|Promise<Error>} - A promise that is resolved with a JSON object or
+   *          rejected with the error.
+   */
+  loadJson (url) {
+    return fetch(url).then(response => response.json())
+  }
+}
+
+// TODO: Shall probably move this to data models
+CedictData.characterForms = {
+  SIMPLIFIED: 'simplified',
+  TRADITIONAL: 'traditional'
+}
```

Comment:
I think I've seen this in Irina's branch but it has not been merged to master yet, I believe.

Reply:
I just checked the latest master version of data-models and I did not find any constants for simplified and traditional forms there. I think Chinese language model should probably expose them. Am I looking at the wrong place?

Reply:
I didn't put these values to Constants of the data-models.
It defines directly inside chinese-source.js

Reply:
> My idea was that a client adapter will take a word and a context and build a list of possible word combinations form it: Ch0, Ch0Ch1, Ch0Ch1Ch2, ... Ch0Ch1...ChN, where Ch is a single Chinese character (the zero one is the one selected by user and Ch1...ChN are the ones from the context forward) and N is a number of characters in context forward. Then client adapter will send a list of words to the data object and will receive a response object back.
> 
> Why this way? I think it is a responsibility of a client adapter (or maybe even a lexical query object) to build a list of possible words out of a context. This is business logic and it does not belong to the data object. On my opinion, data object is a "keeper" of data and its single purpose must be to return requested data back efficiently. The data object is a lower level entity and it should have no knowledge of higher-level concepts such as context forward. @irina060981, @balmas, what do you think?
> 

I agree with this.

Reply:
Ah, I remember this discussion now. We agreed to keep these values out of Constants for now. Not sure what the best final approach is here.